py/foundation/entity.py

```diff
@@ -71,31 +38,25 @@ def words(self) -> Iterable['Word']:
 
     STRONGLY RECOMMENDED not to override this.
     """
-    yield from chain.from_iterable(e.words() for e in self.children)
+    yield from chain.from_iterable(E.words() for E in self.children)
+
+  @property
+  def entity_text(self) -> Optional[str]:
+    entity_fields = (field.name for field in fields(self))
+    return self.text if 'text' in entity_fields else None # type: ignore # how to??
 
 
 @dataclass(frozen=True)
 class Page(Entity):
-  """ A Page is defined by an image region, or region in a document.
+  """A Page is defined by an image region, or region in a document.
 
   Its bounding box is its dimensions, translated by its offset within the
   document. For example, a document with 3 pages, each 50x100, "stacking"
   pages on top of each other would put page 2's bbox as:
       top_left: (0, 100), bottom_right: (50, 200)
   """
   index: int
-  bbox: BBox
-
-  @staticmethod
-  def from_proto(msg: PbEntityPayloadType) -> 'Page':
-    assert isinstance(msg, entity_pb2.Page)
-    index = msg.index
-    bbox = unwrap(BBox.from_proto(msg.bbox))
-    return Page(index, bbox)
-
-  def to_proto(self) -> entity_pb2.Page:
-    msg = entity_pb2.Page(index=self.index, bbox=self.bbox.to_proto())
-    return msg
+  type: str = 'Page'
```

Comment:
Just to confirm, so now everything has a `type` field, correct? And this is included when serializing to JSON?

Reply:
Yep